File: DevExpress.OutlookInspiredApp.Wpf/CS/DevExpress.OutlookInspiredApp.Wpf/Behaviors/ThemeSelectorBehavior.cs
using DevExpress.Xpf.Ribbon;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using DevExpress.Xpf.Core;
using System.Windows.Data;
using DevExpress.Xpf.Bars;

namespace DevExpress.DevAV.Common.View {
    public class ThemeSelectorBehavior : RibbonGalleryItemThemeSelectorBehavior {
        protected override ICollectionView CreateCollectionView() {
            ICollectionView view = CollectionViewSource.GetDefaultView(
                Theme.Themes.Where(t => t.ShowInThemeSelector
            && !t.Name.EndsWith(ThemeManager.TouchDelimiterAndDefinition, StringComparison.InvariantCultureIgnoreCase)
            && !t.Name.StartsWith("Office2013", StringComparison.InvariantCultureIgnoreCase)
            && !(t.Name.StartsWith("Office2016", StringComparison.InvariantCultureIgnoreCase) && !t.Name.EndsWith("SE", StringComparison.InvariantCulture)))
            .Select(t => new ThemeViewModel(t)).ToArray());
            view.GroupDescriptions.Add(new PropertyGroupDescription("Theme.Category"));
            return view;
        }
    }
}

File: DevExpress.OutlookInspiredApp.Wpf/VB/DevExpress.OutlookInspiredApp.Wpf/Behaviors/ThemeSelectorBehavior.vb
Imports DevExpress.Xpf.Ribbon
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel
Imports DevExpress.Xpf.Core
Imports System.Windows.Data
Imports DevExpress.Xpf.Bars

Namespace DevExpress.DevAV.Common.View
    Public Class ThemeSelectorBehavior
        Inherits RibbonGalleryItemThemeSelectorBehavior

        Protected Overrides Function CreateCollectionView() As ICollectionView
            Dim view As ICollectionView = CollectionViewSource.GetDefaultView(Theme.Themes.Where(Function(t) t.ShowInThemeSelector AndAlso (Not t.Name.EndsWith(ThemeManager.TouchDelimiterAndDefinition, StringComparison.InvariantCultureIgnoreCase)) AndAlso (Not t.Name.StartsWith("Office2013", StringComparison.InvariantCultureIgnoreCase)) AndAlso Not(t.Name.StartsWith("Office2016", StringComparison.InvariantCultureIgnoreCase) AndAlso (Not t.Name.EndsWith("SE", StringComparison.InvariantCulture)))).Select(Function(t) New ThemeViewModel(t)).ToArray())
            view.GroupDescriptions.Add(New PropertyGroupDescription("Theme.Category"))
            Return view
        End Function
    End Class
End Namespace
